source/server/admin/stats_request.cc

```diff
@@ -124,19 +124,26 @@ void StatsRequest::populateStatsForCurrentPhase(const ScopeVec& scope_vec) {
 }
 
 template <class StatType> void StatsRequest::populateStatsFromScopes(const ScopeVec& scope_vec) {
-  for (const Stats::ConstScopeSharedPtr& scope : scope_vec) {
-    Stats::IterateFn<StatType> fn = [this](const Stats::RefcountPtr<StatType>& stat) -> bool {
-      if (params_.used_only_ && !stat->used()) {
-        return true;
-      }
-      std::string name = stat->name();
-      if (params_.filter_.has_value() && !std::regex_search(name, params_.filter_.value())) {
+  Stats::IterateFn<StatType> check_stat = [this](const Stats::RefcountPtr<StatType>& stat) -> bool {
+    if (params_.used_only_ && !stat->used()) {
```

Comment:
Maybe for a future PR, but can we combine the filtering logic between here and prometheus? It appears to be the same logic.

Reply:
Yes -- there's a comment in the Prometheus code in the PR explaining why they differ a bit. You could make them share code but it will either make the code more complex or will compromise performance in one call-site or the other.

Reply:
clarified comments in both places.